data-prepper-plugins/cloudwatch-metrics-source/src/main/java/org/opensearch/dataprepper/plugins/source/CloudwatchMetricsWorker.java

```diff
@@ -0,0 +1,198 @@
+                            .value(dimensionsListConfig.getDimensionConfig().getValue()).build());
+                }
+                try {
+                    Metric met = Metric.builder()
+                            .metricName(metricDataQueriesConfig.getMetricsConfig().getName())
+                            .namespace(namespacesListConfig.getNamespaceConfig().getName())
+                            .dimensions(dimensionCollection)
+                            .build();
+
+                    MetricStat metStat = MetricStat.builder()
+                            .stat(metricDataQueriesConfig.getMetricsConfig().getStat())
+                            .unit(metricDataQueriesConfig.getMetricsConfig().getUnit())
+                            .period(metricDataQueriesConfig.getMetricsConfig().getPeriod())
+                            .metric(met)
+                            .build();
+
+                    MetricDataQuery dataQuery = MetricDataQuery.builder()
+                            .metricStat(metStat)
+                            .id(metricDataQueriesConfig.getMetricsConfig().getId())
+                            .returnData(true)
+                            .build();
+
+                    List<MetricDataQuery> dataQueries = new ArrayList<>();
+                    dataQueries.add(dataQuery);
+
+                    GetMetricDataRequest getMetReq = GetMetricDataRequest.builder()
+                            .maxDatapoints(10)
+                            .scanBy(ScanBy.TIMESTAMP_DESCENDING)
+                            .startTime(Instant.parse(namespacesListConfig.getNamespaceConfig().getStartTime()))
+                            .endTime(Instant.parse(namespacesListConfig.getNamespaceConfig().getEndTime()))
+                            .metricDataQueries(dataQueries)
+                            .build();
+
+                    GetMetricDataResponse response = cloudWatchClient.getMetricData(getMetReq);
+                    List<MetricDataResult> data = response.metricDataResults();
+                    writeToBuffer(data, bufferAccumulator, null);
+
+                } catch (CloudWatchException | DateTimeParseException ex) {
+                    LOG.error("Exception Occurred while scraping the metrics  {0}", ex);
+                }
+            }
+        }
+    }
+
+    /**
+     * Helps to write metrics data to buffer and to send end to end acknowledgements after successful processing
+     * @param metricsData metricsData
+     * @param bufferAccumulator bufferAccumulator
+     * @param acknowledgementSet acknowledgementSet
+     */
+    public void writeToBuffer(final List<MetricDataResult> metricsData,
+                              final BufferAccumulator<Record<Event>> bufferAccumulator,
+                              final AcknowledgementSet acknowledgementSet) {
+        metricsData.forEach(message -> {
+            final Record<Event> eventRecord = new Record<Event>(JacksonEvent.fromMessage(message.toString()));
```

Comment:
You are creating a string from the metrics. This is not very useful for pipeline authors. You need to convert the CloudWatch metrics into a `org.opensearch.dataprepper.model.metric.Metric` instance. This is an `Event` type already, so you can wrap it in a `Record` and save it.

This is a very important aspect of the source and it won't be very useful with it.

Also, we'd need unit tests for this. So, you should do all that translation in a new class that is dedicated to the translation of a CloudWatch metric to a Data Prepper `Metric` class.

Reply:
@dlvenable - Cloudwatch metrics are in format https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_GetMetricData.html i.e. sample reponse

{
  "MetricDataResults": [
    {
      "Id": "m1",
      "StatusCode": "Complete",
      "Label": "CPUUtilization, peak of 31.5 was at 1-22 13:05",
      "Timestamps": [
        1518868032,      
        1518867732,
        1518867432
      ],
      "Values": [
        15000,
        14000,
        16000
      ]
    },
}

which does not fit in to 'org.opensearch.dataprepper.model.metric.Metric'  instance object structure.  Should Event object be enhanced or a new Object to be created for CloudWatch in Metric ?